libs/langchain/langchain/document_loaders/parsers/docai.py

```diff
@@ -0,0 +1,292 @@
+
+from langchain.docstore.document import Document
+from langchain.document_loaders.base import BaseBlobParser
+from langchain.document_loaders.blob_loaders import Blob
+from langchain.utils.iter import batch_iterate
+
+if TYPE_CHECKING:
+    from google.api_core.operation import Operation
+    from google.cloud.documentai import DocumentProcessorServiceClient
+
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class DocAIParsingResults:
+    """A dataclass to store DocAI parsing results."""
+
+    source_path: str
+    parsed_path: str
+
+
+class DocAIParser(BaseBlobParser):
+    def __init__(
+        self,
+        *,
+        client: Optional["DocumentProcessorServiceClient"] = None,
+        location: Optional[str] = None,
+        gcs_output_path: Optional[str] = None,
+        processor_name: Optional[str] = None,
+    ):
+        """Initializes the parser.
+
+        Args:
+            client: a DocumentProcessorServiceClient to use
+            location: a GCP location where a DOcAI parser is located
+            gcs_output_path: a path on GCS to store parsing results
+            processor_name: name of a processor
+
+        You should provide either a client or location (and then a client
+            would be instantiated).
+        """
+        if client and location:
+            raise ValueError(
+                "You should provide either a client or a location but not both "
+                "of them."
+            )
+        if not client and not location:
+            raise ValueError(
+                "You must specify either a client or a location to instantiate "
+                "a client."
+            )
+
+        self._gcs_output_path = gcs_output_path
+        self._processor_name = processor_name
+        if client:
+            self._client = client
+        else:
+            try:
+                from google.api_core.client_options import ClientOptions
+                from google.cloud.documentai import DocumentProcessorServiceClient
+            except ImportError:
+                raise ImportError(
+                    "documentai package not found, please install it with"
+                    " `pip install google-cloud-documentai`"
+                )
+            options = ClientOptions(
+                api_endpoint=f"{location}-documentai.googleapis.com"
+            )
+            self._client = DocumentProcessorServiceClient(client_options=options)
+
+    def lazy_parse(self, blob: Blob) -> Iterator[Document]:
+        """Parses a blob lazily.
+
+        Args:
+            blobs: a Blob to parse
+
+        This is a long-running operations! A recommended way is to batch
+            documents together and use `batch_parse` method.
+        """
+        yield from self.batch_parse([blob], gcs_output_path=self._gcs_output_path)
+
+    def batch_parse(
+        self,
+        blobs: Sequence[Blob],
+        gcs_output_path: Optional[str] = None,
+        timeout_min: int = 60,
```

Comment:
is this supposed to be 60 minutes? also any reason to have check_in in seconds and timeout in minutes? imo would be easier if they were in the same units

Reply:
OK, all in sec now